### PR TITLE
Add group create processor

### LINKED ISSUE #23544: Process `Group.CREATE` command

### PR DESCRIPTION
**Pull request by koevskinikola**
## Description

Add a `GroupCreateProcessor` and test coverage for it.

## Checklist

- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)

## Related issues

closes #23544

### CODE REVIEW

**Comment by koevskinikola** [comment]
@remcowesterhoud @nicpuppa feel free to delegate to someone else if you don't have the time to review this PR.

File: zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/GroupTest.java [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+    final var createdGroup = groupRecord.getValue();
+    Assertions.assertThat(createdGroup).isNotNull().hasFieldOrPropertyWithValue("name", name);
+
+    assertThat(duplicatedGroupRecord)
+        .hasRejectionType(RejectionType.ALREADY_EXISTS)
+        .hasRejectionReason(
+            "Expected to create group with name '"
+                + name
+                + "', but a group with this name already exists");
```
Comment (nicpuppa) [inline zeebe/engine/src/test/java/io/camunda/zeebe/engine/processing/authorization/GroupTest.java:51]: 🔧 

```
"Expected to create group with name '%s', but a group with this name already exists."
                .formatted(name)
```

File: zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/GroupCreateProcessor.java [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+      rejectionWriter.appendRejection(command, RejectionType.UNAUTHORIZED, errorMessage);
+      responseWriter.writeRejectionOnCommand(command, RejectionType.UNAUTHORIZED, errorMessage);
+      return;
+    }
+
+    final var record = command.getValue();
+    final var groupKey = groupState.getGroupKeyByName(record.getName());
+    if (groupKey.isPresent()) {
+      final var errorMessage =
+          "Expected to create group with name '%s', but a group with this name already exists"
```
Comment (nicpuppa) [inline zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/identity/GroupCreateProcessor.java:71]: 🔧 Adding a dot at the end of the error message
```
Expected to create group with name '%s', but a group with this name already exists.
```

**Review by nicpuppa — APPROVED** [review]
Just some small suggestions, overall great work 👍

**Pull request merged**